scraper/src/fcc2zim/build.py

```diff
@@ -0,0 +1,138 @@
+
+    redirects = []
+    for superblock in superblock_dict:
+        course_list = superblock_dict[superblock]
+        for course in course_list:
+            meta_json_path = pathlib.Path(
+                curriculum_dist_dir,
+                "curriculum",
+                fcc_lang,
+                superblock,
+                course,
+                "_meta.json",
+            )
+            challenges = json.loads(meta_json_path.read_text())["challenges"]
+            for challenge in challenges:
+                title = challenge["title"]
+                redirects.append(
+                    (f'{fcc_lang}/{superblock}/{course}/{challenge["slug"]}', title)
+                )
+
+    return OrderedDict(redirects).items()
+
+
+def build_command(
+    zimui_dist_dir: pathlib.Path,
+    out_dir: pathlib.Path,
+    language: str,
+    name: str,
+    title: str,
+    description: str,
+    long_description: str | None,
+    creator: str,
+    publisher: str,
+    curriculum_dist_dir: pathlib.Path,
+    zim_file: str | None,
+    *,
+    force: bool,
+):
+    Global.logger.info("Scraper: build phase starting")
+
+    period = datetime.today().strftime("%Y-%m")  # noqa: DTZ002
+    if zim_file:
+        zim_path = pathlib.Path(zim_file.format(period=period))
+        # make sure we were given a filename and not a path
+        if pathlib.Path(zim_path.name) != zim_path:
+            raise ValueError(f"zim_name is not a filename: {zim_file}")
+    else:
+        zim_path = pathlib.Path(f"{name}_{period}.zim")
+
+    # build full path
+    zim_path = out_dir.joinpath(zim_path)
+
+    if zim_path.exists():
+        if not force:
+            raise ValueError(f"ZIM file {zim_path} already exist.")
+        Global.logger.info(f"Removing existing ZIM file {zim_path}")
+        zim_path.unlink()
+    else:
+        Global.logger.info(f"ZIM path: {zim_path}")
+
+    with Creator(zim_path, "index.html").config_metadata(
+        Name=name,
+        Title=title,
+        Publisher=publisher,
+        Date=datetime.now(),  # noqa: DTZ005
+        Creator=creator,
+        Description=description,
+        LongDescription=long_description,
+        Language=language,
+        Tags=";".join(["FCC", "freeCodeCamp"]),
+        Scraper=f"fcc2zim v{VERSION}",
+        Illustration_48x48_at_1=logo_path.read_bytes(),
+    ) as creator:
+        # Add zimui files
+        for file in zimui_dist_dir.rglob("*"):
+            if file.is_dir():
+                continue
+            if file.suffix == ".map":
+                Global.logger.debug(f"IGNORING: {file}")
+                continue
+            path = str(pathlib.Path(file).relative_to(zimui_dist_dir))
+            Global.logger.debug(f"Adding {path} to ZIM")
+            creator.add_item_for(path, fpath=file)
+
+        for file in curriculum_dist_dir.rglob("*"):
+            if file.is_dir():
+                continue
+            if file.suffix == ".map":
```

Comment:
Are there such files there?

Reply:
Not from my experience ... I kept it just because I did not knew why it was here to avoid a regression. I can get rid of both if ok for you

Reply:
I originally had sourcemaps turned on in Vite, this can be removed as long as it's not re-enabled.

Reply:
`.map` files are ignored twice: once when adding the client and this second time when adding the curriculum. I would not expect curriculum files to have source maps (but haven't checked the actual files)

Reply:
Yeah, then that's definitely a mistake on my part, curriculum files should only be .json and .md.

Reply:
The fact that they are ignored twice is "my fault", I split the processing of "zimui" and the processing of "prebuilt fcc data" ; since I did not knew where the map files came from, I kept the code that was ignoring `.map` files in both places. Now that it is clear we do not need it at all, I will remove both.